src/transformers/models/modernbert/modular_modernbert.py

```diff
@@ -1422,7 +1449,7 @@ def forward(
         )
         last_hidden_state = outputs[0]
 
-        last_hidden_state = self.drop(last_hidden_state)
+        last_hidden_state = self.head(last_hidden_state, attention_mask, apply_pooling=False)
```

Comment:
if it is always false, why do we have an argument for it?

Reply:
The pooling layer needs to pools for heads which need it, like `ModernBertForSequenceClassification`, and not pool for heads which don't, like `ModernBertForTokenClassification`.

I can swap `ModernBertPoolingHead` out for the `ModernBertPredictionHead` in the `ModernBertForTokenClassification`, if you want.

I thought this way would be less confusing and easier to follow when/if future ForTask heads get added